tutorials/20_REST_API.ipynb

```diff
@@ -0,0 +1,485 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "docker-compose haystack-api"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Without Docker\n",
+    "\n",
+    "If you prefer to not use Docker with your Haystack API, you can start the REST API server and supporting Haystack pipeline by running the gunicorn server manually with the following:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "pip install --upgrade pip\n",
+    "pip install 'farm-haystack[all]' ## or 'all-gpu' for the GPU-enabled dependencies\n",
+    "pip install -e rest_api/\n",
+    "\n",
+    "brew install xpdf  ## required for `PDFToTextConverter` \n",
+    "\n",
+    "gunicorn rest_api.application:app -b 0.0.0.0:8000 -k uvicorn.workers.UvicornWorker -t 300 ## start the gunicorn server"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can find more information about installing Haystack at [the installation guide](https://docs.haystack.deepset.ai/docs/installation)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "When setting up is done, you should notice that: \n",
+    "* Haystack API: listens on port 8000\n",
+    "* DocumentStore (Elasticsearch): listens on port 9200\n",
+    "\n",
+    "Test whether everything is okay by going to Swagger documentation for the Haystack REST API on [`http://127.0.0.1:8000/docs`](http://127.0.0.1:8000/docs) and trying out `/initialized` endpoint or sending a cURL request as `curl -X GET http://127.0.0.1:8000/initialized`. \n",
+    "\n",
+    "If everything is alright, you can start asking questions! Wikipedia pages about countries and capital are already indexed to Elasticsearch by the docker image we provided in [`docker-compose.yml`](https://github.com/deepset-ai/haystack/blob/main/docker-compose.yml#L22). To ask questions, you can use `/query` endpoint again via Haystack REST API UI or a cURL request.   "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "curl -X 'POST' \\\n",
+    "  'http://127.0.0.1:8000/query' \\\n",
+    "  -H 'accept: application/json' \\\n",
+    "  -H 'Content-Type: application/json' \\\n",
+    "  -d '{\n",
+    "  \"query\": \"What is the capital of Sudan\",\n",
+    "  \"params\": {}\n",
+    "}'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create your pipeline\n",
+    "\n",
+    "Now you know how REST API works, you can start customizing the pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Start Elasticsearch \n",
```

Comment:
To be honest I found this section a bit confusing and I think it could be clearer. Why do they have to change the ES image? 

Also I think they could do with more explanation as to how to restart Elasticsearch - let's give them a specific command to run.

I also think telling them to change a specific line by number is dangerous - if a line is added or removed, this number is no longer valid.